src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -726,6 +710,65 @@ public void SetDefaultProjectIndex(int selectedIndex)
             {
                 _solutionManager.Value.DefaultNuGetProjectName = null;
             }
+
+            StartAsyncDefaultProjectUpdate();
+        }
+
+        private void StartAsyncDefaultProjectUpdate()
+        {
+            Assumes.Present(_solutionManager.Value);
+
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                // Ensure _solutionExistsCookie has been assigned.
+                await _vsMonitorSelection.GetValueAsync();
+
+                var hr = VsMonitorSelection.IsCmdUIContextActive(_solutionExistsCookie, out var pfActive);
+                bool isInvalidationRequired;
+
+                if (!(ErrorHandler.Succeeded(hr) && pfActive > 0))
+                {
+                    isInvalidationRequired = DefaultProject != null;
+
+                    DefaultProject = null;
+
+                    if (isInvalidationRequired)
+                    {
+                        CommandUiUtilities.InvalidateDefaultProject();
+                    }
+
+                    return;
+                }
+
+                NuGetProject project = await _solutionManager.Value.GetDefaultNuGetProjectAsync();
```

Comment:
This does need to be on the UI does it?

Reply:
I do not need to call `SwitchToMainThreadAsync()` before this.  Internally, during initialization `VSSolutionManager` may call it though.

Reply:
I mean to say, does not need to be. 

I see earlier we were doing. 
UI thread -> back to threadpool.

Normally that's not necessarily any better since the method in question can switch to the UI thread if it needs to.